### PR TITLE
Enable TIAF for Python Tests as a Jenkins job

### PR DESCRIPTION
**Pull request by jcurtisk**
## What does this PR do?

This PR adds support for the newly created TIAF Python Runtime within the Python scripts for TIAF. It also enables the Python TIAF jenkins job, so that monitoring of test selection can be fed back into reporting services. This job is non-gating and has a negligible run time(1-5s).
## How was this PR tested?

Manual Testing backed up by the Automated unit tests

### CODE REVIEW

File: scripts/build/TestImpactAnalysis/test_impact/base_test_impact.py [hunk 1]
```diff
@@ -390,6 +434,37 @@ def _compile_s3_top_level_dir_name(self, dir_name: str):
...
+        test_runs += report[UNEXECUTED_TEST_RUNS]
+        return test_runs
+
+    def _extract_test_runs_from_sequence_report(self, report):
+        """
+        Extract all relevant test runs from the report generated by the TIAF runtime.
+        @param report: The report generated by the TIAF runtime, in dict format.
+        """
+        report_type = report['type']
+        test_runs = self._extract_test_runs_from_test_run_report(report['selected_test_run_report'])
```
Comment (jonawals) [inline scripts/build/TestImpactAnalysis/test_impact/base_test_impact.py:462]: Can these string literals be replaced with named constants? They might already be in the MARS script.

File: scripts/build/TestImpactAnalysis/test_impact/base_test_impact.py [hunk 2]
```diff
@@ -390,6 +434,37 @@ def _compile_s3_top_level_dir_name(self, dir_name: str):
...
+
+        if report_type == "impact_analysis" or report_type == "safe_impact_analysis":
+            test_runs = test_runs + self._extract_test_runs_from_test_run_report(report["drafted_test_run_report"])
```
Comment (jonawals) [inline scripts/build/TestImpactAnalysis/test_impact/base_test_impact.py:465]: Can these string literals be replaced with named constants? They might already be in the MARS script.

File: scripts/build/TestImpactAnalysis/test_impact/base_test_impact.py (hunk 1 above)
Comment (jcurtisk) [reply scripts/build/TestImpactAnalysis/test_impact/base_test_impact.py]: Done

File: scripts/build/TestImpactAnalysis/test_impact/base_test_impact.py (hunk 2 above)
Comment (jcurtisk) [reply scripts/build/TestImpactAnalysis/test_impact/base_test_impact.py]: Done

**Review by jonawals — APPROVED** [review]

File: scripts/build/TestImpactAnalysis/tiaf_report_constants.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+DISCARDED_TEST_RUNS_KEY = "discarded_test_runs"
+INSTRUMENTATION_KEY = "instrumentation"
+EFFICIENCY_KEY = "efficiency"
+CONFIG_KEY = "config"
+POLICY_KEY = "policy"
+CHANGE_LIST_KEY = "change_list"
+TEST_RUN_SELECTION_KEY = "test_run_selection"
+DYNAMIC_DEPENDENCY_MAP_POLICY_KEY = "dynamic_dependency_map"
+DYNAMIC_DEPENDENCY_MAP_POLICY_UPDATE_KEY = "update"
+REPORT_KEY = "report"
```
Comment (jonawals) [inline scripts/build/TestImpactAnalysis/tiaf_report_constants.py:83]: Missing new line at EOF.

File: scripts/build/Platform/Windows/build_config.json [hunk 4]
```diff
@@ -45,7 +46,7 @@
       "SCRIPT_PARAMETERS": "--platform=Windows --repository=%REPOSITORY_NAME% --jobname=%JOB_NAME% --jobnumber=%BUILD_NUMBER% --jobnode=%NODE_LABEL% --changelist=%CHANGE_ID%"
     }
   },
-  "test_impact_analysis_profile": {
+  "test_impact_analysis_profile_native": {
```
Comment (amzn-changml) [inline scripts/build/Platform/Windows/build_config.json:49]: Just confirming: Do we intend to run this in AR? It's not on any pipeline steps or contain a suffix in the name to allow it to run nightly.

**Review by amzn-changml — APPROVED** [review]
Looks reasonable, with just a minor comment

File: scripts/build/Platform/Windows/build_config.json (hunk 4 above)
Comment (jcurtisk) [reply scripts/build/Platform/Windows/build_config.json:49]: Yes. We are not intending to run `test_impact_analysis_native` in AR at this time.

**Pull request merged**